components/sync15/README.md

```diff
@@ -5,4 +5,108 @@ data stores that want to sync against a Firefox Sync v1.5 sync server.
 It handles things like encrypting/decrypting records, obtaining and
 using storage node auth tokens, and so-on.
 
-It also needs a more complete README...
+There are 2 key concepts to understand here - the implementation itself, and
+a rust trait for a "syncable store" where component-specific logic lives - but
+before we dive into them, some preamble might help put things into context.
+
+## Introduction and History
+
+For many years Sync has worked exclusively against a "sync v1.5 server". This
+[is a REST API described here](https://mozilla-services.readthedocs.io/en/latest/storage/apis-1.5.html).
+The important part is that the API is conceptually quite simple - there are
+arbitrary "collections", indexed by a GUID, and lacking traditonal database
+concepts like joins. Because the payload is encrypted, there's very little
```

Comment:
nit: "payload" sounds like it might be a technical term that new readers may worry that they're already supposed to understand; something like "the stored items are encrypted" may be better.